cli/deployment/config.go

```diff
@@ -370,7 +370,6 @@ func Config(flagset *pflag.FlagSet, vip *viper.Viper) (*codersdk.DeploymentConfi
 		return nil, xerrors.Errorf("get global config from flag: %w", err)
 	}
 	vip.SetEnvPrefix("coder")
-	vip.AutomaticEnv()
```

Comment:
Have you reviewed open issues for the [spf13/viper](https://github.com/spf13/viper/issues)? There are a few open regarding `vip.AutomaticEnv`. Maybe it's a bug you could raise with the team.

Reply:
Not really, I thought it'd be good for us to manually control these (we already had all the code in place to do it). But it's a good suggestion, I did a quick browse but nothing stood out. I also just pushed a follow-up PR for this: https://github.com/coder/coder/pull/4894 in support of not enabling `AutomaticEnv`.

Reply:
This might be related, but it's not the exact same issue. Still, looks like a rabbit hole I don't want to dive into 😄 https://github.com/spf13/viper/issues/1243

Reply:
Good for me, thanks for reviewing those issues! I suspected that this might be a rabbit hole hence the idea of covering it with a basic unit test. It's up to you to decide if you prefer to open a new issue for viper or leave it as it :)